ranger/fs.go

```diff
@@ -0,0 +1,73 @@
+package ranger
```

Comment:
This is where the magic is: a custom implementation of `fs.FS`. The files for `trails` are embedded whereas OS files are treated like a regular filesystem. You can have, essentially, multiple file trees with the same filepaths but different files served up. It prioritizes OS-level files (so the files in a trails project like `college-try`, etc.) over files made available in `trails` itself.